jerry-core/ecma/builtin-objects/ecma-builtin-array-prototype.c

```diff
@@ -605,6 +589,18 @@ ecma_builtin_array_prototype_object_reverse (ecma_value_t this_arg, /**< this ar
     }
 #endif /* ENABLED (JERRY_ES2015_BUILTIN_PROXY) */
 
+    bool lower_exist = ecma_is_value_true (has_lower);
+
+    if (lower_exist)
+    {
+      lower_value = ecma_op_object_get (obj_p, lower_str_p);
+
+      if (ECMA_IS_VALUE_ERROR (lower_value))
+      {
+        goto clean_up;
```

Comment:
In these error cases, should we return the error for the user? Do we have test cases for these?

Reply:
These test cases have already been tested in the ES5 tests, this update just changes the order of methods, but the behaviour when one of these error occurs is the same.